### PR TITLE
fix issue with #15077

### PR DESCRIPTION
**Pull request by vtjnash**
still needs a test, but wanted to at least push the diff for now

### CODE REVIEW

**Comment by JeffBezanson** [comment]
+1

File: base/dict.jl [hunk 1]
```diff
@@ -400,6 +400,10 @@ SerializationState(io::IO) = SerializationState{typeof(io)}(io)
 
 # dict
 
+# These can be changed, to trade off better performance for space
+const global maxallowedprobe = [16]
+const global maxprobeshift   = [6]
```
Comment (tkelman) [inline base/dict.jl]: should these use `Ref` ?

Comment (JeffBezanson) [inline base/dict.jl]: No, these should just be constants.

Comment (StefanKarpinski) [inline base/dict.jl]: Agree. If anything, setting these on a per-dict choice makes more sense. Having global parameters like this means that to solve one particular performance problem in one part of the system, you end up changing the behavior of every dict everywhere in the system.

File: test/dict.jl [hunk 2]
```diff
@@ -461,9 +475,55 @@ let badKeys = ASCIIString["FINO_emv5.0","FINO_ema0.1","RATE_ema1.0","NIBPM_ema1.
...
+                     0xd02c,0x862d,0x8f34,0xe529,0xf938,0x4f39,0xd03a,0x473b,0x1e3b,0x1d3a,
+                     0xcc39,0x7339,0xcf40,0x8740,0x813d,0xe640,0xc443,0x6344,0x3744,0x2c3d,
+                     0x8c48,0xdf49,0x5743]
+
+    # Walk through all possible hash values (mod size of hash table)
+    for offset = 0:1023
+        d2 = Dict{MyInt, Int}()
+        hashoffset[] = offset
+        for i = 1:length(badKeys)
+            d2[MyInt(badKeys[i])] = i
```
Comment (tkelman) [inline test/dict.jl]: failing when inlining is off, evidently? https://build.julialang.org/builders/coverage_ubuntu14.04-x64/builds/91/steps/Run%20non-inlined%20tests/logs/stdio

**Pull request merged**